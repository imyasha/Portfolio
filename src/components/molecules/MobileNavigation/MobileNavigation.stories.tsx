import { MobileNavigation } from './MobileNavigation';

export default {
	title: 'molecules/MobileNavigation',
	component: MobileNavigation,
};

export const MobileNavigationStory = () => <MobileNavigation />;

MobileNavigationStory.storyName = 'MobileNavigation';
MobileNavigationStory.args = {};
