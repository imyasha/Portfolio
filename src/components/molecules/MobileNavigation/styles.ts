import styled from 'styled-components';

const StyledMobileNavigation = styled.ul`
	display: flex;

	li {
		width: 2rem;
		margin-left: 1rem;
	}
`;

const Menu = styled.button`
	width: 2rem;
	height: 2rem;
	padding: 0;
	background: transparent;
	border: 0;
	color: ${({ theme }) => theme.text};
	position: absolute;
	z-index: 100;

	&:hover {
		color: ${({ theme }) => theme.headline};
	}
`;

const Overlay = styled.div`
	position: absolute;
	width: 100%;
	height: 100%;
	top: -50%;
	left: 0;
	background: ${({ theme }) => theme.body};
	display: flex;
	align-items: center;
	justify-content: center;
	opacity: 0;
	pointer-events: none;
	transition: all 0.5s ease;
	z-index: 10;

	&.open {
		opacity: 1;
		pointer-events: auto;
		top: 0;
	}
`;

export { Menu, Overlay, StyledMobileNavigation };
