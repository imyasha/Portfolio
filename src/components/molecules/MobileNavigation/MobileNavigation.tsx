import classnames from 'classnames';
import { FC, useState } from 'react';
import { Menu, Overlay, StyledMobileNavigation } from './styles';

import { Icon } from 'Atoms/Icon';
import { Navigation } from 'Molecules/Navigation';
import { ThemeToggle } from 'Atoms/ThemeToggle';

const MobileNavigation: FC = () => {
	const [isOpen, setOpen] = useState(false);

	const toggle = () => setOpen((prev) => !prev);

	return (
		<>
			<StyledMobileNavigation>
				<li>
					<ThemeToggle />
				</li>
				<li>
					<Menu onClick={toggle}>
						<Icon icon={isOpen ? 'CLOSE' : 'MENU'} />
					</Menu>
				</li>
			</StyledMobileNavigation>
			<Overlay
				className={classnames({
					open: isOpen,
				})}
			>
				<Navigation onClick={() => setOpen(false)} />
			</Overlay>
		</>
	);
};

export { MobileNavigation };
