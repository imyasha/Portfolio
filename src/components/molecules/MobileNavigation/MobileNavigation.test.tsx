import { MobileNavigation } from '../MobileNavigation';
import { render } from '@testing-library/react';

const mockDispatch = jest.fn();

jest.mock('react-redux', () => ({
	useSelector: jest.fn(),
	useDispatch: () => mockDispatch,
}));

describe('MobileNavigation', () => {
	it('renders correctly', () => {
		const { container } = render(<MobileNavigation />);
		expect(container).toMatchSnapshot();
	});
});
