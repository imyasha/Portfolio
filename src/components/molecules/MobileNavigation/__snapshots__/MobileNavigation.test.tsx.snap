// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileNavigation renders correctly 1`] = `
<div>
  <ul
    class="styles__StyledMobileNavigation-sc-14uzdzl-0 crjHjf"
  >
    <li>
      <button
        class="styles__StyledThemeToggle-sc-1dfmkyd-0 iOJryI"
        title="Switch to light theme"
      >
        <test-file-stub />
      </button>
    </li>
    <li>
      <button
        class="styles__Menu-sc-14uzdzl-1 dJJTrF"
      >
        <test-file-stub />
      </button>
    </li>
  </ul>
  <div
    class="styles__Overlay-sc-14uzdzl-2 grizif"
  >
    <nav
      class="styles__StyledNavigation-sc-16epmp-0"
    >
      <ul
        class="styles__List-sc-16epmp-1 gPSCMv"
      >
        <li>
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="/about"
          >
            About
          </a>
        </li>
        <li>
          <a
            href="/uses"
          >
            Uses
          </a>
        </li>
        <li>
          <a
            href="/blog"
          >
            Blog
          </a>
        </li>
        <li>
          <a
            href="/portfolio"
          >
            Portfolio
          </a>
        </li>
        <li
          class="desktop-only"
        >
          <button
            class="styles__StyledThemeToggle-sc-1dfmkyd-0 iOJryI"
            title="Switch to light theme"
          >
            <test-file-stub />
          </button>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;
